src/dune/build_system.ml

```diff
@@ -361,6 +362,47 @@ let set_rule_generators ~init ~gen_rules =
   Fdecl.set t.init_rules init_rules;
   Fdecl.set t.gen_rules gen_rules
 
+let set_vcs vcs =
+  let open Fiber.O in
+  let t = t () in
+  let () = Fdecl.set t.vcs vcs in
+  match t.caching with
+  | None -> Fiber.return ()
+  | Some ({ cache = (module Caching); _ } as caching) ->
+    let+ caching =
+      let+ with_repositories =
+        let f ({ Vcs.root; _ } as vcs) =
+          let+ commit = Vcs.commit_id vcs in
+          { Cache.directory = Path.to_absolute_filename root
+          ; remote = "FIXME: not filled yet"
+          ; commit
+          }
+        in
+        let+ repositories = Fiber.parallel_map ~f (Fdecl.get t.vcs) in
```

Comment:
I find it rather confusing that we are using two names "repository" and "vcs" to mean the same thing. I think we should choose one and stick to it throughout the codebase, including the communication protocol while we are relatively free to change it. I think "repository" is a better name since it combines two pieces of information: a path and a particular VCS type (`Git` or `Hg`).

I would suggest to leave a TODO in the module `vcs.mli` to rename it to `repository.mli`. and use "repository" where possible in this module.

@jeremiedimino What do you think about renaming `vcs.mli` to `repository.mli` for consisteny with the Dune protocol? I'm happy to do the actual renaming.